src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -519,3 +519,28 @@ def raise_unsupported_error_for_flex_vmss(vmss, error_message):
             and vmss.orchestration_mode.lower() == 'flexible':
         from azure.cli.core.azclierror import ArgumentUsageError
         raise ArgumentUsageError(error_message)
+
+
+def is_trusted_launch_supported(supported_features):
+    if not supported_features:
+        return False
+
+    trusted_launch = {'TrustedLaunchSupported', 'TrustedLaunch', 'TrustedLaunchAndConfidentialVmSupported'}
+
+    return bool(trusted_launch.intersection({feature.value for feature in supported_features}))
+
+
+def trusted_launch_warning_log(namespace, generation_version, features):
+    if not generation_version:
+        return
+
+    if generation_version == 'V1':
+        logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS/DISK image'
+                       ' and Trusted Launch security type. To know more about Trusted Launch, please visit'
+                       ' https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
```

Comment:
My original idea was to recommend users to deploy Gen 2 images when using Gen 1 images. Otherwise, users may not know that the Trusted Launch only is supported by Gen 2.  @AjKundnani What do you think of this?

Reply:
@zhoxing-ms - The link to [Trusted Launch VM Page](https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch) includes details about Gen2 VMs and link to page for Gen2 VMs as well. if the user navigates to link, they'll get the information about Gen2 VMs.

Reply:
OK, make sense